src/popup/router/components/BoxButton.vue

```diff
@@ -1,31 +1,44 @@
 <template>
-  <a
-    v-if="isExternalLink"
-    :class="['box-button', fill ]"
-    :href="to"
-    target="_blank"
-  >
-    <slot />
-  </a>
-  <RouterLink
-    v-else
-    :class="['box-button', fill ]"
+  <Component
+    :is="component"
+    class="box-button"
+    :class="[
+      fill,
+      {
+        'new-ui': newUi,
+      },
+    ]"
     :to="to"
+    :href="href"
+    :target="(href) ? '_blank' : null"
+    v-on="$listeners"
   >
     <slot />
-  </RouterLink>
+  </Component>
 </template>
 
 <script>
 export default {
   props: {
-    to: { type: [String, Object], default: '' },
+    to: { type: [String, Object], default: null },
+    href: { type: String, default: null },
     fill: {
       type: String,
       validator: (value) => ['primary', 'alternative'].includes(value),
       default: 'primary',
     },
-    isExternalLink: { type: Boolean },
+    newUi: Boolean,
+  },
+  computed: {
+    component() {
+      if (this.to) {
+        return 'RouterLink';
+      }
+      if (this.href) {
+        return 'a';
+      }
+      return 'button';
+    },
```

Comment:
component() {
      if (this.to) return 'RouterLink';
      if (this.href) return 'a'
      return 'button';
    },

Reply:
Writing the conditional statements this way are more error prone and from my perspective looks bad, but this is rather subjective. I prefer longer conditional blocks as this is the default setting for eslint in many cases.

If teams want to achieve strict rules for the code the eslint is always set for "curly" rule to error.